### PR TITLE
feat: add spring boot example which interacts with blob storage

### PR DESCRIPTION
**Pull request by JasonTheDeveloper**
**Reason for Change**:
This PR introduces a new sample Spring Boot project to help demonstrate how to interact with Azure Storage using a mange identity. 

With this sample you can:
* `POST` request and the body of that request will be saved to a container in Azure Storage.
* `GET` request with the file name of the blob specified as a parameter will return the contents of the file.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [ ] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [ ] ran `make precommit`

**Issue Fixed**:
ref https://github.com/Azure/aad-pod-identity/issues/153

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

Yes, it does. The sample is based heavily on docs found here: https://docs.microsoft.com/en-us/azure/developer/java/spring-framework/configure-spring-boot-starter-java-app-with-azure-storage.

I haven't notified the maintainer - there are about 5 contributors on that page so not sure who I should reach out to.

I have linked it at the top of `README.md`.

**Notes for Reviewers**:
The `README.md` should explain everything.

### CODE REVIEW

File: examples/java-blob/src/main/java/com/microsoft/cse/storageexample/controller/BlobController.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+		BlobClient blobClient = containerClient.getBlobClient(fileName);
+		blobClient.download(stream);
+
+		return stream.toString();
+	}
+
+	@PostMapping
+	public String writeBlobFile(@RequestBody String data) throws Exception {
+		String fileName = "quickstart-" + java.util.UUID.randomUUID() + ".txt";
```
Comment (chewong) [inline examples/java-blob/src/main/java/com/microsoft/cse/storageexample/controller/BlobController.java:42]:
```suggestion
		String fileName = String.format("quickstart-%s.txt", java.util.UUID.randomUUID());
```

File: examples/java-blob/src/main/java/com/microsoft/cse/storageexample/controller/BlobController.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+		BlobClient blobClient = containerClient.getBlobClient(fileName);
+		blobClient.upload(new ByteArrayInputStream(data.getBytes(StandardCharsets.UTF_8)), data.length());
+
+		return "file " + fileName + " was uploaded";
```
Comment (chewong) [inline examples/java-blob/src/main/java/com/microsoft/cse/storageexample/controller/BlobController.java:47]:
```suggestion
		return String.format("file %s was uploaded", fileName);
```

File: examples/java-blob/README.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
+# Spring Boot + Azure Storage + AAD Pod Identity
+
+This tutorial is based on [this documentation](https://docs.microsoft.com/en-us/azure/developer/java/spring-framework/configure-spring-boot-starter-java-app-with-azure-storage).
+
+This repository contains sample code and instructions demonstrating how you can access Azure Storage using a user assigned manage identity with the aid of AAD Pod Identity from a spring boot application.
```
Comment (chewong) [inline examples/java-blob/README.md:5]:
```suggestion
This repository contains sample code and instructions demonstrating how you can access Azure Storage using a user-assigned identity with the aid of AAD Pod Identity from a spring boot application.
```

File: examples/java-blob/README.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+This sample takes a `POST` request and saves in as a text file in an Azure Blob Storage container.
```
Comment (chewong) [inline examples/java-blob/README.md:7]:
```suggestion
This sample takes a `POST` request and saves it as a text file in an Azure Blob Storage container.
```

File: examples/java-blob/README.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+* AAD Pod Identity installed and configured
+* Azure Container Registry
+* Azure Storage
+    * Make sure you have already created a container
+
+## Setup
+
+### Create Manage Identity and Assign Role
+
+In this step we'll create a new manage identity which will be used to interact with the Azure Storage Account.
```
Comment (chewong) [inline examples/java-blob/README.md:21]:
```suggestion
In this step, we'll create a new user-assigned identity which will be used to interact with the Azure Storage account.
```

File: examples/java-blob/README.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    AKS_MSI_CLIENTID=$(az aks show \
+        -n $AKS_NAME \
+        -g $AKS_RESOURCEGROUP \
+        --query "identityProfile.kubeletidentity.clientId" \
+        -o tsv)
+
+    STORAGE_ACCOUNT_RESOURCEID=$(az storage account show \
+        --name $STORAGE_ACCOUNT_NAME \
+        --query 'id' -o tsv)
+    ```
```
Comment (chewong) [inline examples/java-blob/README.md:46]: Could you rename the environment variables to those used in https://azure.github.io/aad-pod-identity/docs/demo/standard_walkthrough/?

**Review by chewong — CHANGES_REQUESTED** [review]
Thank you so much for this. For a more consistent demo experience across the repository, could you move the markdown file to https://github.com/Azure/aad-pod-identity/tree/master/website/content/en/docs/Demo so it gets rendered as a web page?

File: examples/java-blob/README.md (hunk 6 above)
Comment (JasonTheDeveloper) [reply examples/java-blob/README.md]: No problem! Done.

File: examples/java-blob/README.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+This tutorial is based on [this documentation](https://docs.microsoft.com/en-us/azure/developer/java/spring-framework/configure-spring-boot-starter-java-app-with-azure-storage).
+
+This repository contains sample code and instructions demonstrating how you can access Azure Storage using a user-assigned identity with the aid of AAD Pod Identity from a spring boot application.
+
+This sample takes a `POST` request and saves it as a text file in an Azure Blob Storage container.
+
+## Prerequisites
+
+* AKS cluster
```
Comment (chewong) [inline examples/java-blob/README.md:11]:
```suggestion
* an AKS cluster with [managed identity enabled](https://docs.microsoft.com/en-us/azure/aks/use-managed-identity)
```

File: examples/java-blob/README.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+In this step, we'll create a new user-assigned identity which will be used to interact with the Azure Storage account.
+
+**Note:** this step can be skipped if you already have a manage identity you'd like to reuse.
+
+1. Setup:
+
+    ```sh
+    CLUSTER_NAME=<YOUR_AKS_CLUSTER_NAME>
+    CLUSTER_RESOURCEGROUP=$(az aks list --query "[?name == '$CLUSTER_NAME'].resourceGroup" -o tsv)
```
Comment (chewong) [inline examples/java-blob/README.md:29]:
```suggestion
    CLUSTER_RESOURCE_GROUP=$(az aks list --query "[?name == '$CLUSTER_NAME'].resourceGroup" -o tsv)
```

File: examples/java-blob/README.md [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+    STORAGE_ACCOUNT_NAME=<STORAGE_ACCOUNT_NAME>
+    STORAGE_ACCOUNT_RESOURCEGROUP=$(az storage account list --query "[?name == '$STORAGE_ACCOUNT_NAME'].resourceGroup" -o tsv)
+
+    CLUSTER_MSI_CLIENTID=$(az aks show \
+        -n $CLUSTER_NAME \
+        -g $CLUSTER_RESOURCEGROUP \
+        --query "identityProfile.kubeletidentity.clientId" \
+        -o tsv)
+
+    STORAGE_ACCOUNT_RESOURCEID=$(az storage account show \
```
Comment (chewong) [inline examples/java-blob/README.md:43]:
```suggestion
    STORAGE_ACCOUNT_RESOURCE_ID=$(az storage account show \
```

File: examples/java-blob/README.md [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+
+    ```sh
+    CLUSTER_NAME=<YOUR_AKS_CLUSTER_NAME>
+    CLUSTER_RESOURCEGROUP=$(az aks list --query "[?name == '$CLUSTER_NAME'].resourceGroup" -o tsv)
+
+    IDENTITY_NAME=<YOUR_IDENTITY_NAME>
+    IDENTITY_RESOURCE_GROUP=<YOUR_RESOURCE_GROUP>
+
+    STORAGE_ACCOUNT_NAME=<STORAGE_ACCOUNT_NAME>
+    STORAGE_ACCOUNT_RESOURCEGROUP=$(az storage account list --query "[?name == '$STORAGE_ACCOUNT_NAME'].resourceGroup" -o tsv)
```
Comment (chewong) [inline examples/java-blob/README.md:35]:
```suggestion
    STORAGE_ACCOUNT_RESOURCE_GROUP=$(az storage account list --query "[?name == '$STORAGE_ACCOUNT_NAME'].resourceGroup" -o tsv)
```

File: examples/java-blob/README.md [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+**Note:** this step can be skipped if you already have a manage identity you'd like to reuse.
+
+1. Setup:
+
+    ```sh
+    CLUSTER_NAME=<YOUR_AKS_CLUSTER_NAME>
+    CLUSTER_RESOURCEGROUP=$(az aks list --query "[?name == '$CLUSTER_NAME'].resourceGroup" -o tsv)
+
+    IDENTITY_NAME=<YOUR_IDENTITY_NAME>
+    IDENTITY_RESOURCE_GROUP=<YOUR_RESOURCE_GROUP>
```
Comment (chewong) [inline examples/java-blob/README.md:32]: For simplicity, we can use the node resource group to store the user-assigned identity:
```suggestion
    IDENTITY_RESOURCE_GROUP=$(az aks show -g $CLUSTER_RESOURCE_GROUP -n $CLUSTER_NAME --query nodeResourceGroup -otsv)
```

File: examples/java-blob/README.md [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+
+* AKS cluster
+* AAD Pod Identity installed and configured
+* Azure Container Registry
+* Azure Storage
+    * Make sure you have already created a container
+
+## Setup
+
+### Create Manage Identity and Assign Role
```
Comment (chewong) [inline examples/java-blob/README.md:19]:
```suggestion
### Create a Managed Identity and Assign Roles
```

File: examples/java-blob/README.md [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+    ```sh
+    CONTAINER=test
+
+    az role assignment create \
+        --role "Storage Blob Data Contributor" \
+        --assignee $IDENTITY_CLIENT_ID \
+        --scope "$(STORAGE_ACCOUNT_RESOURCEID)/blobServices/default/containers/$(CONTAINER)"
+    ```
+
+    **Note**: if you want the manage identity to access your entire Storage Account, you can ignore `/blobServices/default/containers/$(CONTAINER)`.
```
Comment (chewong) [inline examples/java-blob/README.md:82]:
```suggestion
    **Note**: if you want the managed identity to access your entire Storage Account, you can ignore `/blobServices/default/containers/$(CONTAINER)`.
```

File: examples/java-blob/README.md [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+        --resource-group $IDENTITY_RESOURCE_GROUP \
+        --query 'id' -o tsv)
+
+    IDENTITY_CLIENT_ID=$(az identity show \
+        -n $IDENTITY_NAME \
+        -g $IDENTITY_RESOURCE_GROUP \
+        --query 'clientId' -o tsv)
+    ```
+
+3. Assign role to allow AAD Pod Identity to access our newly created manage identity:
```
Comment (chewong) [inline examples/java-blob/README.md:62]:
```suggestion
3. Assign role to allow AAD Pod Identity to access our newly created managed identity:
```

File: examples/java-blob/README.md [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+
+4. Grant permission to specific container in the Azure Storage Account:
+
+    ```sh
+    CONTAINER=test
+
+    az role assignment create \
+        --role "Storage Blob Data Contributor" \
+        --assignee $IDENTITY_CLIENT_ID \
+        --scope "$(STORAGE_ACCOUNT_RESOURCEID)/blobServices/default/containers/$(CONTAINER)"
```
Comment (chewong) [inline examples/java-blob/README.md:79]:
```suggestion
        --scope "$STORAGE_ACCOUNT_RESOURCE_ID/blobServices/default/containers/$CONTAINER"
```

File: examples/java-blob/README.md [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+    ```
+
+    **Note**: if you want the manage identity to access your entire Storage Account, you can ignore `/blobServices/default/containers/$(CONTAINER)`.
+
+### Configure AAD Pod Identity
+
+The following step will create a new `AzureIdentity` resource in Kubernetes in the `blob` namespace. If your namespace is something different then change NAMESPACE to match the namespace you want to deploy into.
+
+```sh
+NAMESPACE=blob
```
Comment (chewong) [inline examples/java-blob/README.md:89]:
```suggestion
NAMESPACE=blob
kubectl create namespace $NAMESPACE
```

File: examples/java-blob/README.md [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+Now we'll create the `AzureIdentityBinding` and specify the selector.
+
+```sh
+POD_LABEL_SELECTOR=IDENTITY_NAME
+
+cat << EOF | kubectl apply -f -
+apiVersion: aadpodidentity.k8s.io/v1
+kind: AzureIdentityBinding
+metadata:
+  name: $(IDENTITY_NAME)-binding
```
Comment (chewong) [inline examples/java-blob/README.md:113]:
```suggestion
  name: $IDENTITY_NAME-binding
```

File: examples/java-blob/README.md [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+  type: 0
+  resourceID: $IDENTITY_RESOURCE_ID
+  clientID: $IDENTITY_CLIENT_ID
+EOF
+```
+
+Now we'll create the `AzureIdentityBinding` and specify the selector.
+
+```sh
+POD_LABEL_SELECTOR=IDENTITY_NAME
```
Comment (chewong) [inline examples/java-blob/README.md:107]:
```suggestion
POD_LABEL_SELECTOR=$IDENTITY_NAME
```

File: examples/java-blob/README.md [hunk 19]
```diff
@@ -0,0 +1,215 @@
...
+    CLUSTER_NAME=<YOUR_AKS_CLUSTER_NAME>
+    CLUSTER_RESOURCEGROUP=$(az aks list --query "[?name == '$CLUSTER_NAME'].resourceGroup" -o tsv)
+
+    IDENTITY_NAME=<YOUR_IDENTITY_NAME>
+    IDENTITY_RESOURCE_GROUP=<YOUR_RESOURCE_GROUP>
+
+    STORAGE_ACCOUNT_NAME=<STORAGE_ACCOUNT_NAME>
+    STORAGE_ACCOUNT_RESOURCEGROUP=$(az storage account list --query "[?name == '$STORAGE_ACCOUNT_NAME'].resourceGroup" -o tsv)
+
+    CLUSTER_MSI_CLIENTID=$(az aks show \
```
Comment (chewong) [inline examples/java-blob/README.md:37]:
```suggestion
    CLUSTER_MSI_CLIENT_ID=$(az aks show \
```

**Review by chewong — CHANGES_REQUESTED** [review]

**Comment by chewong** [comment]
For a more consistent demo experience across the repository, could you move `examples/java-blob/README.md` to https://github.com/Azure/aad-pod-identity/tree/master/website/content/en/docs/Demo so it gets rendered as a web page in https://azure.github.io/aad-pod-identity/docs/demo/?

**Review by chewong — APPROVED** [review]
Left some final comments but LGTM, I can help open a subsequent PR to move the demo to the website directory. Thank you once again for working on this!

File: examples/java-blob/README.md [hunk 20]
```diff
@@ -0,0 +1,216 @@
...
+* Azure Storage
+    * Make sure you have already created a container
+
+## Setup
+
+### Create a Managed Identity and Assign Roles
+
+In this step, we'll create a new user-assigned identity which will be used to interact with the Azure Storage account.
+
+**Note:** this step can be skipped if you already have a manage identity you'd like to reuse.
```
Comment (aramase) [inline examples/java-blob/README.md:23]: nit: s/manage/managed

File: examples/java-blob/README.md [hunk 21]
```diff
@@ -0,0 +1,216 @@
...
+        -g $CLUSTER_RESOURCE_GROUP \
+        --query "identityProfile.kubeletidentity.clientId" \
+        -o tsv)
+
+    STORAGE_ACCOUNT_RESOURCE_ID=$(az storage account show \
+        --name $STORAGE_ACCOUNT_NAME \
+        --query 'id' -o tsv)
+    ```
+
+2. Create the manage identity:
```
Comment (aramase) [inline examples/java-blob/README.md:48]: nit: s/manage/managed

**Review by aramase — APPROVED** [review]
Thank you for the PR. 

Just few small nits, but we can hash those out in a follow up PR.

**Pull request merged**